kuma/users/templatetags/jinja_helpers.py

```diff
@@ -132,6 +134,24 @@ def provider_login_url(context, provider_id, **params):
     return Markup(provider.get_login_url(request, **params))
 
 
+@library.global_function
+@contextfunction
+def remote_login_url(context, **params):
+    """Redirects to the read-only site and makes sure the 'next' query
+    string is fully domained."""
+    request = context['request']
+    next = None
+    if 'next' not in params:
+        next = get_request_param(request, 'next')
+    if not next:
+        next = request.get_full_path()
```

Comment:
Lines 143-147 don't seem right to me (although it all works in the end due to lines 146-147). You're never using the `next` keyword argument if one is provided. It seems like you'd want something like this instead:
```python
next = params.get('next') or get_request_param(request, 'next') or request.get_full_path()
```